src/execution/StatusManager.ts

```diff
@@ -90,39 +95,32 @@ export default class StatusManager {
   }
 
   /**
-   * Create and push a new process into the execution. If a process for the given id already exists the existing process is set to PENDING instead.
-   * @param  {String} id  Identifier for the process. Used to identify already existing processes.
+   * Create and push a new process into the execution.
+   * @param  {ProcessType} type Type of the process. Used to identify already existing processes.
    * @param  {Step} step The step that should contain the new process.
-   * @param  {ProcessMessage} message  A ProcessMessage for this Process. Will be used on newly created or already existing process.
-   * @param  {object} [params]   Additional parameters to append to the process.
+   * @param  {Status} status By default created procces is set to the STARTED status. We can override new process with the needed status.
    * @return {Process}
    */
   findOrCreateProcess = (
-    id: string,
+    type: ProcessType,
     step: Step,
-    message: ProcessMessage,
-    params?: OptionalParameters
+    status?: Status
   ): Process => {
     if (!step.execution || !step.execution.process) {
-      throw new Error('Execution has not been initialized')
+      throw new Error("Execution hasn't been initialized.")
     }
 
-    const process = step.execution.process.find((p) => p.id === id)
+    const process = step.execution.process.find((p) => p.type === type)
 
     if (process) {
       return process
     }
 
     const newProcess: Process = {
-      id: id,
+      type: type,
       startedAt: Date.now(),
-      message: message,
-      status: 'PENDING',
-    }
-    if (params) {
-      for (const [key, value] of Object.entries(params)) {
-        newProcess[key] = value
-      }
+      message: getProcessMessage(type, status ?? 'STARTED'),
+      status: status ?? 'STARTED',
```

Comment:
You are defaulting to `STARTED` if no status is given. But in `processMessages` not every type has a `STARTED` value. Wouldn't that be a problem?